samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -992,8 +999,14 @@ int sock_send_data(
 		}
 
 	} else if (data_out != NULL && data_out_length > 0) {
+		/* Set requested blocking mode for duration of data sending */
+		set_sock_blocking_mode(socket_info->fd, blocking);
+
 		/* Send data if it's given and is not zero length */
-		sock_send(socket_info, data_out, data_out_length, true, data_format_hex);
+		sock_send(socket_info, data_out, data_out_length, flags, true, data_format_hex);
```

Comment:
Do you need to pass the flags as a separate argument when they are stored already in `socket_info->send_flags`?

Reply:
No, I will remove this extra argument.